test/DurableTask.Sdk.Tests/Grpc/DurableTaskGrpcClientIntegrationTests.cs

```diff
@@ -0,0 +1,151 @@
+using Xunit.Abstractions;
+
+namespace Microsoft.DurableTask.Grpc.Tests;
+
+public class DurableTaskGrpcClientIntegrationTests : IntegrationTestBase
+{
+    const string OrchestrationName = "TestOrchestration";
+
+    public DurableTaskGrpcClientIntegrationTests(ITestOutputHelper output, GrpcSidecarFixture sidecarFixture)
+        : base(output, sidecarFixture)
+    {
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task GetInstanceMetadata_EndToEnd(bool shouldThrow)
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, string instanceId, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().Be(instanceId);
+                metadata.SerializedInput.Should().BeNull();
+                metadata.SerializedOutput.Should().BeNull();
+                metadata.RuntimeStatus.Should().Be(status);
+                metadata.FailureDetails.Should().BeNull();
+            }
+        }
+
+        await using DurableTaskGrpcWorker server = await this.StartAsync();
+        DurableTaskClient client = this.CreateDurableTaskClient();
+
+        string instanceId = await client.ScheduleNewOrchestrationInstanceAsync(OrchestrationName, input: shouldThrow);
+
+        await client.WaitForInstanceStartAsync(instanceId, default);
+        OrchestrationMetadata? metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, OrchestrationRuntimeStatus.Running);
+
+        await client.RaiseEventAsync(instanceId, "event", default);
+        await client.WaitForInstanceCompletionAsync(instanceId, default);
+        metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, shouldThrow ? OrchestrationRuntimeStatus.Failed : OrchestrationRuntimeStatus.Completed);
+    }
+
+    [Fact]
+    public async Task GetInstances_EndToEnd()
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().StartWith("GetInstances_EndToEnd-");
+                metadata.RuntimeStatus.Should().Be(status);
+
+                // InMemoryOrchestrationService always returns these in a query.
+                // metadata.FailureDetails.Should().BeNull();
+                // metadata.SerializedInput.Should().BeNull();
+                // metadata.SerializedOutput.Should().BeNull();
+            }
+        }
+
+        OrchestrationQuery query = new() { InstanceIdPrefix = "GetInstances_EndToEnd" };
+
+        static async Task ForEachAsync(Func<string, Task> func)
```

Comment:
nit: based on the name, I mistakenly thought this was a generic For-Each implementation, but I now see it's hardcoded to two orchestration instances. Maybe name this `ForEachOrchestrationAsync`, or something less generic?